python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -331,22 +309,25 @@ def __getitem__(self, this_id):
             this_id (int): This is the key of an object in the object dictionary
         """
         logger.debug("__getitem__")
+        # Is this an Incomplete Object?
+        if this_id in self._incomplete_objects:
+            return IncompleteObject(self, this_id)
+
+        # Nope, so try to find it and raise an exception if not
         try:
             return self._objects[this_id]
         except KeyError as err:
             logger.debug("Repo KeyError: %s" % err)
             logger.debug("Keys: %s id: %s" % (self._objects.keys(), this_id))
-            if this_id in self._incomplete_objects:
```

Comment:
OK - I think I get it. So there is a point where an object can be referenced in **both** `_objects` and `_incomplete_objects` and in this case it should return the given object rather than an `IncompleteObject`. If that's the case I'll make a note and revert the code back.